subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -949,7 +949,18 @@ void ull_conn_done(struct node_rx_event_done *done)
 	 * Slave received terminate ind or MIC failure
 	 */
 	reason_peer = conn->llcp_terminate.reason_peer;
-	if (reason_peer && (lll->role || conn->master.terminate_ack)) {
+	if (reason_peer && (
```

Comment:
@cvinayak , this seems a bit overly complicated to follow. I propose the following alternative:

```
tmp = 1;
#if CONFIG_BT_CENTRAL
tmp = conn->master.terminate_ack;
#endif
#if CONFIG_BT_PERIPHERAL		    
tmp ||= lll->role;
#endif

if (reason_peer && tmp) {
	...
}
```

Does this make sense?